bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/ContentAccessHandler.java

```diff
@@ -392,22 +396,46 @@ public Response doGet( final String packageType, final String type, final String
                     {
                         return handleMissingContentQuery( sk, path, builderModifier );
                     }
+                    else if ( packageType.equals( NPM_PKG_KEY ) )
```

Comment:
I'm a little confused about this part. How does it differ from the normal retrieval code? It seems like the only difference is that it's reading the package.json string and passing that back as the entity, instead of streaming it back. We stream pom.xml content back in maven use cases, so it's safe to stream package.json content that way too.

What does this extra conditional logic do that the default logic doesn't? 

If we truly need to customize doGet functionality by package type, then we should add a Function parameter or something to customize it from the package-specific module, instead of adding a dependency to the package-specific module into the core classes. I know I asked you to use the ContentAccessHandler directly, but I'd like us to find a way to do it without putting direct dependencies on all of our package type implementations.